storage/postgres/postgres.go

```diff
@@ -378,6 +385,7 @@ WHERE
   (? = '' OR operation_name = ?) AND
   DURATION(time) >= ? AND
   DURATION(time) <= ? AND
+  EXISTS ( SELECT 1 FROM spans AS sub_spans WHERE sub_spans.trace_id = spans.trace_id AND sub_spans.service_name IN (` + strings.Join(serviceConds, ", ") + `)) AND
```

Comment:
The API allows it; and even if the contract didn't, we do not want to generate invalid SQL on a possible code path. We should either support it, or detect and reject it.

The PostgreSQL documentation documents IN as `expression IN (value [, ...])`, which would suggest that an empty IN is not valid.